### PR TITLE
feat:[SSCA-659]: Added Mirgrations for Old Artifact and EnforcementSummary Entities

### PR DESCRIPTION
**Pull request by arpit-harness**
## Describe your changes
Onboarded the migration framework and added two scripts for migration of artifact and enforcement summary entities.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54473)

### CODE REVIEW

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ logging:
       stackdriverLogEnabled: false
 
 mongo:
-  uri: mongodb://localhost:27017/ng-harness
+  uri: mongodb://mongo1:30001,mongo2:30002,mongo3:30003/ng-harness
```
Comment (inderschera) [inline 120-ng-manager/config.yml:90]: revert these changes

File: 360-cg-manager/config.yml [hunk 2]
```diff
@@ -308,8 +308,8 @@ dmsSecret: IC04LYMBf1lDP5oeY4hupxd4HJhLmN6azUku3xEbeE3SUx5G3ZYzhbiwVtK4i7AmqyU9O
 
 
 mongo:
-  uri: mongodb://localhost:27017/harness
-  #uri: mongodb://mongo1:30001,mongo2:30002,mongo3:30003
+  #uri: mongodb://localhost:27017/harness
+  uri: mongodb://mongo1:30001,mongo2:30002,mongo3:30003/harness
```
Comment (inderschera) [inline 360-cg-manager/config.yml:312]: revert these changes

File: ssca-manager/service/src/main/java/io/harness/ssca/migration/ArtifactEntityMigration.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+      query = new Query(Criteria.where(NormalizedSBOMEntityKeys.accountId)
+                            .is(artifact.getAccountId())
+                            .and(NormalizedSBOMEntityKeys.orgIdentifier)
+                            .is(artifact.getOrgId())
+                            .and(NormalizedSBOMEntityKeys.projectIdentifier)
+                            .is(artifact.getProjectId())
+                            .and(NormalizedSBOMEntityKeys.orchestrationId)
+                            .is(artifact.getOrchestrationId()));
+      Long count = Long.valueOf(sbomComponentRepo.findAllByQuery(query).size());
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/ssca/migration/ArtifactEntityMigration.java:73]: just run the count query here instead of getting the list and then getting the size.

File: ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryCreatedTimeMigration.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    Criteria criteria = Criteria.where(EnforcementSummaryEntityKeys.createdAt).is(null);
+    Query query = new Query(criteria);
+    FindIterable<Document> iterable =
+        mongoTemplate.getCollection(mongoTemplate.getCollectionName(EnforcementSummaryEntity.class))
+            .find(query.getQueryObject());
+
+    for (Document document : iterable) {
+      EnforcementSummaryEntity summaryEntity =
+          mongoTemplate.getConverter().read(EnforcementSummaryEntity.class, document);
+      summaryEntity.setCreatedAt(currentTime);
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryCreatedTimeMigration.java:45]: Do we store timestamp in enforcement summary? Can we get timestamp from there?

Comment (arpit-harness) [reply ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryCreatedTimeMigration.java:45]: No we don't have timestamps in either EnforcementResult or EnforcementSummary.

File: ssca-manager/service/src/main/java/io/harness/ssca/migration/ArtifactEntityMigration.java (hunk 3 above)
Comment (arpit-harness) [reply ssca-manager/service/src/main/java/io/harness/ssca/migration/ArtifactEntityMigration.java]: Done.

**Comment by arpit-harness** [comment]
trigger uts

**Review by inderschera — APPROVED** [review]
